src/rhsm/connection.py

```diff
@@ -1361,7 +1361,7 @@ def regenEntitlementCertificates(self, consumer_id, lazy_regen=True):
             result = True
         except (RemoteServerException, httpslib.BadStatusLine) as e:
             # 404s indicate that the service is unsupported (Candlepin too old, or SAM)
-            if isinstance(e, httpslib.BadStatusLine) or e.code == "404":
+            if isinstance(e, httpslib.BadStatusLine) or str(e.code) == "404":
```

Comment:
I would split this into two except clauses, one for RemoteServerException and one for httpslib.BadStatusLine

But what is causing a BadStatusLine to bubble up this far?  Seems like that should be handled at a lower level  (Restlib._request?) and raising a rhsm exception (rhsm.connection.RemoteServerException or maybe NetworkException). Then this method would only need to special case RemoteServerException with a 404 here.

(or really, for httplib.BadStatusLine since M2Crypto.httpslib.BadStatusLine is httplib.BadStatusLine imported into M2Crypto.httpslib's namespace)